server/routes/api/user.js

```diff
@@ -1,7 +1,47 @@
 const express = require('express');
-
 const router = express.Router();
+const bcrypt = require('bcrypt');
+
+// User Model
+const User = require('../../models/user');
+
+// Test User route
+router.get('/test', (req, res) => res.json({ msg: 'User route works!' }));
+
+// POST User Registration
+router.post('/register', (req, res) => {
+  const email = req.body.email.trim();
+  const password = req.body.password.trim();
+
+  User.findOne({ email: email }).then(user => {
+    if (user) {
+      return res.status(400).json({ email: 'Email already exists.' });
```

Comment:
This should be 'You must provide an email address.' We'll want to stick exactly to the requirements, because that's what we'll be testing for.

Reply:
Added - ` if (!email)
    res.status(400).json({ email: 'You must provide an email address.' });`

I didn't see the language for this in the issue.  "Email already exists." is for the uniqueness issue.  Also no language, but think we should return an error for non-unique email.  Thoughts?

Reply:
Oops, I misread that. This line is fine. Also, I just remembered:

```
.save()
          .then(user =>
            res.status(201).json({
              id: user._id,
              user: user.email,
              createdAt: user.createdAt
            })
          )
          .catch(err => res.status(400).json({ error: err }));
```

^^ this should send back a required error (or possibly invalid email). Let's lean on that for any empty emails. You can remove the line you just added. Sorry for the confusion!